python/packaging/frontend_sdist/setup.py

```diff
@@ -15,42 +15,96 @@
 # limitations under the License.
 #
 
+import os
+import subprocess
 import sys
 
 from setuptools import setup
 from setuptools.command.install import install
-import subprocess as sp
 
 tensorrt_module = "##TENSORRT_MODULE##"
+tensorrt_version = "##TENSORRT_PYTHON_VERSION##"
+tensorrt_submodules = [
+    "{}_libs=={}".format(tensorrt_module, tensorrt_version),
+    "{}_bindings=={}".format(tensorrt_module, tensorrt_version),
+]
 
 
 class InstallCommand(install):
     def run(self):
-        def install_dep(package_name):
-            status = sp.run(
-                [
-                    sys.executable,
-                    "-m",
-                    "pip",
-                    "install",
-                    "{:}==##TENSORRT_PYTHON_VERSION##".format(package_name),
-                    "--index-url",
-                    "https://pypi.nvidia.com",
-                ]
-            )
-            status.check_returncode()
-
-        install_dep("{:}_libs".format(tensorrt_module))
-        install_dep("{:}_bindings".format(tensorrt_module))
-
-        install.run(self)
+        # pip-inside-pip hack ref #3080
+        subprocess.check_call(
+            [
+                "{}/bin/pip".format(sys.exec_prefix),
+                "install",
+                "--extra-index-url",
+                "https://pypi.nvidia.com",
+                *tensorrt_submodules,
+            ]
+        )
+
+        super().run()
+
+
+def pip_config_list():
+    """Get the current pip config (env vars, config file, etc)."""
+    return subprocess.check_output(
+        [
+            "{}/bin/pip".format(sys.exec_prefix),
+            "config",
+            "list",
+        ]
+    ).decode()
+
+
+def parent_command_line():
+    """Get the command line of the parent PID."""
+    pid = os.getppid()
+    # try retrieval using psutil
+    try:
+        import psutil
+
+        return " ".join(psutil.Process(pid).cmdline())
+    except ModuleNotFoundError:
+        pass
+    # fall back to shell
+    try:
+        return (
+            subprocess.check_output(["ps", "-p", str(pid), "-o", "command"])
```

Comment:
Adding `--no-headers` avoids the need for the `split()`. Checking both CentOS 7 and Ubuntu 18.04 containers this option has been supported since at least then and TRT supports those OSes and newer.

Reply:
e89d6b5